src/EventTimer.ts

```diff
@@ -66,8 +74,32 @@ export class EventTimer {
 		return this.init();
 	}
 
+	/**
+	 * Returns all commercial timers. CMP-related timers are not tracked
+	 * by EventTimer so they need to be concatenated to EventTimer's private events array.
+	 */
+	public get events(): Event[] {
+		return typeof window.performance !== 'undefined' &&
+			'mark' in window.performance
+			? [
+					...this._events,
+					...EventTimer._externallyDefinedEventNames
+						.map(
+							(eventName) =>
+								new Event(
+									eventName,
+									window.performance.getEntriesByName(
+										eventName,
+									)[0] ?? 0,
+								),
+						)
+						.filter((event) => event.ts > 0),
```

Comment:
We could probably be more explicit here and filter when `window.performance.getEntriesByName(eventName)` is empty rather than setting the timestamp to `0` and filter on `event.ts > 0`.

It might be clearer on first glance to see we're removing those with no entries rather than those with `0` timestamps.

Reply:
That's a valid point, and something I had considered. In fact, the original implementation was exactly as you describe, but it broke when I tested it locally because of the way the `Event` constructor worked (it couldn't read `startTime` of `undefined`). But on second thought, this happened because I was `map`ping before `filter`ing. Swapping the order around solved the issue. Tried it locally and it works, will commit in a second.